src/client/Microsoft.Identity.Client/Platforms/net45/NetDesktopDeviceAuthManager.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Permissions;
+using System.Threading.Tasks;
+#if DESKTOP
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Platforms.net45.Native;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.Identity.Client.Platforms.net45
+{
+    internal class NetDesktopDeviceAuthManager : IDeviceAuthManager
+    {
+        public bool CanHandleDeviceAuthChallenge
+        {
+            get { return true; }
+        }
+
+        public async Task<string> CreateDeviceAuthChallengeResponseAsync(IDictionary<string, string> challengeData)
```

Comment:
I do not understand how much code you have written and how much was just copied from ADAL - please, next time you port code, please let reviewers know so that we don't review copied code. 

The issue here are unit tests: 

- if you copied this code, it's likely that you don't understand it enough to write unit tests. Are there unit tests in ADAL around this? If so, they need to be copied. If not, we need to asses the importance of this code, i.e. how likely it is to change. If it is likely to have bugs or to evolve, we need tests. I would ask @henrik-me for an opinion of shipping this without unit tests.

Reply:
unit tests are needed. e2e tests are needed